src/libraries/System.Net.Http.Json/tests/FunctionalTests/HttpClientJsonExtensionsTests.cs

```diff
@@ -355,6 +366,52 @@ public async Task AllowNullRequesturlAsync()
                 });
         }
 
+        [Fact]
+        public async Task AllowNullRequestUrlAsAsyncEnumerable()
+        {
+            await HttpMessageHandlerLoopbackServer.CreateClientAndServerAsync(
+                async (handler, uri) =>
+                {
+                    using (HttpClient client = new HttpClient(handler))
+                    {
+                        client.BaseAddress = uri;
+
+                        static void AssertPeopleEquality(List<Person> actualPeople)
+                        {
+                            for (int i = 0; i < People.WomenOfProgramming.Length; i++)
+                            {
+                                var expected = People.WomenOfProgramming[i];
+                                var actual = actualPeople[i];
+
+                                Person.AssertPersonEquality(expected, actual);
+                            }
+                        }
+
+                        List<Person> people = new List<Person>();
+                        await foreach (Person? person in client.GetFromJsonAsAsyncEnumerable<Person>((string)null))
+                        {
+                            people.Add(Assert.IsType<Person>(person));
+                        }
+
+                        AssertPeopleEquality(people);
+
+                        people = new List<Person>();
+                        await foreach (Person? person in client.GetFromJsonAsAsyncEnumerable<Person>((Uri)null))
```

Comment:
It's kind of weird that you need to explicitly pass a `null` to send a request to the default path. Do the existing `GetFromJsonAsync` methods have the same issue?

Reply:
There are two overloads, and we need to disambiguate them. `(string)null` and `(Uri)null`. Yes, the existing `GetFromJsonAsync` methods are done the same way.